crates/shared/src/sources/balancer_v2/pool_fetching/registry.rs

```diff
@@ -64,6 +68,7 @@ where
     web3: Web3,
     fetcher: Arc<dyn PoolInfoFetching<Factory>>,
     updater: PoolUpdater<Factory>,
+    non_existent_pools: RwLock<HashSet<H256>>,
```

Comment:
Could use https://crates.io/crates/delay_map if we want an easy way to make it forget about missing ids after some time (probably easy to implement ourselves as well)

Reply:
Interesting crate. Looks like the exposed API is all we need for our use case. Will add it to the PR.